okhttp/src/main/java/io/grpc/transport/okhttp/OkHttpClientTransport.java

```diff
@@ -381,7 +431,9 @@ private void onGoAway(int lastKnownStreamId, Status status) {
     for (OkHttpClientStream stream : goAwayStreams) {
       stream.transportReportStatus(status, false, new Metadata.Trailers());
     }
-
+    if (outstandingPing != null) {
+      outstandingPing.failed(status.asException());
```

Comment:
Unlike Netty, this go away isn't guaranteed to be called. stopIfNecessary() looks like an appropriate place.